### ISSUE TITLE
New-MgEntitlementManagementAssignmentPolicy cmdlet

### ISSUE THREAD

**Issue by irwins** [microsoftgraph/msgraph-sdk-powershell#1354]
Hi,

I've exhausted all possibilities trying to figure out  why a previous script broke... The previous module version I used was 1.9.6. It was recently been updated to 1.10.0 , also available on the az devops agents

When I use version 1.10.0 I get the following error
![image](https://user-images.githubusercontent.com/10112589/176432389-29637865-2bb3-4a9a-85f1-77242c30eb03.png)

I did use the beta endpoint. How do I begin to troubleshoot this properly?

Rg./Irwin

**Comment by peombwa** [comment]
Thanks for following up on this.

According to the [API reference](https://docs.microsoft.com/en-us/graph/api/entitlementmanagement-post-assignmentpolicies?view=graph-rest-1.0&tabs=http&viewFallbackFrom=graph-rest-beta), the API/Command is only available in v1.0. Please use `Select-MgProfile v1.0` to use the command.
``` powershell
➜ Find-MgGraphCommand -Command New-MgEntitlementManagementAssignmentPolicy

   APIVersion: v1.0

Command                                     Module              Method URI
-------                                     ------              ------ ---
New-MgEntitlementManagementAssignmentPolicy Identity.Governance POST   /identityGovernance/entitlementManagement/assignmentPolicies
```

Please note that APIs under beta version are subject to change and their use in production apps is not supported. See https://docs.microsoft.com/en-us/graph/api/overview?view=graph-rest-beta.

**Comment by peombwa** [comment]
Yes, it does. v1.0 is the default profile. See https://docs.microsoft.com/en-us/powershell/microsoftgraph/get-started?toc=%2Fgraph%2Ftoc.json&view=graph-powershell-1.0#api-version.

Please see the example at https://docs.microsoft.com/en-us/graph/api/entitlementmanagement-post-assignmentpolicies?view=graph-rest-1.0&tabs=powershell#example-2-create-a-policy-for-users-from-other-organizations-to-request for information on to use the command.

**Comment by irwins** [comment]
So I refactored to use  the v1.0 endpoint. 

I haven't figured out how to add the resource in v1.0  

In beta it's the following cmdlet _Get-MgEntitlementManagementAccessPackageCatalogAccessPackageResource_ 

This doesn't exist in v1.0 so how do I add it the accessPackage?

**Comment by irwins** [comment]
Just for reference... https://docs.microsoft.com/en-us/graph/api/entitlementmanagement-post-accesspackageresourcerequests?view=graph-rest-beta&tabs=powershell#example-4-create-an-accesspackageresourcerequest-for-adding-a-group-as-a-resource

**Comment by peombwa** [comment]
You can mix v1.0 and beta commands by calling `Select-MgProfile beta` before calling beta commands and `Select-MgProfile v1.0` before calling v1.0 commands. See https://docs.microsoft.com/en-us/powershell/microsoftgraph/get-started?toc=%2Fgraph%2Ftoc.json&view=graph-powershell-1.0#api-version.
However, you should consider the following disclaimer before using beta APIs/commands:
> APIs under the /beta version in Microsoft Graph are subject to change. Use of these APIs in production applications is not supported. To determine whether an API is available in v1.0, use the Version selector.

Please open a feature request at https://developer.microsoft.com/en-us/graph/support to have the API supported in v1.0.

We plan to change the API selection model in v2 by providing dedicated commands for v1.0 and beta API versions. See https://github.com/microsoftgraph/msgraph-sdk-powershell/issues/692.

**Comment by irwins** [comment]
Ok thanks for the feedback!
